src/dfx/src/lib/models/canister_id_store.rs

```diff
@@ -35,16 +32,18 @@ pub struct CanisterIdStore {
 impl CanisterIdStore {
     pub const DEFAULT: &'static str = "__default";
 
-    #[context("Failed to load canister id store.")]
-    pub fn for_env(env: &dyn Environment) -> DfxResult<Self> {
-        CanisterIdStore::new(env.get_network_descriptor(), env.get_config())
+    pub fn for_env(env: &dyn Environment) -> Result<Self, CanisterStoreIdError> {
+        CanisterIdStore::new(env.get_network_descriptor(), env.get_config()).map_err(|e| {
+            CanisterStoreIdError::LoadCanisterIdStore {
+                source: Box::new(e),
+            }
+        })
```

Comment:
There's enough context in error returned by `fn new`.  Wrapping it additional context just makes the error larger, without giving more information.
```suggestion
    pub fn for_env(env: &dyn Environment) -> Result<Self, CanisterStoreIdError> {
        CanisterIdStore::new(env.get_network_descriptor(), env.get_config())
```